### PR TITLE
Fix issue #75

### PR DESCRIPTION
**Pull request by Angeschossen**
This pull requests fixes issue #75. 
The ProtectionManager class now requires the plugin instance that uses dough.

### CODE REVIEW

File: dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/LandsProtectionModule.java [hunk 1]
```diff
@@ -18,15 +18,16 @@
 public class LandsProtectionModule implements ProtectionModule {
 
     private LandsIntegration landsIntegration;
-    private final Plugin plugin;
+    private final Plugin lands, plugin;
```
Comment (WalshyDev) [inline dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/LandsProtectionModule.java:21]:
```suggestion
    private final Plugin lands;
    private final Plugin plugin;
```

File: dough-protection/src/main/java/io/github/bakedlibs/dough/protection/ProtectionManager.java [hunk 2]
```diff
@@ -50,30 +50,32 @@
      *
      * @param server
      *            The Server your plugin is running on.
+     * @param plugin The plugin that uses dough.
+     *
```
Comment (WalshyDev) [inline dough-protection/src/main/java/io/github/bakedlibs/dough/protection/ProtectionManager.java:54]:
```suggestion
     * @param plugin
     *            The plugin that uses dough.
```

**Review by WalshyDev — CHANGES_REQUESTED** [review]
Appreciate the PR, tiny fixes

**Review by Sfiguz7 — APPROVED** [review]
LGTM

File: dough-protection/src/main/java/io/github/bakedlibs/dough/protection/ProtectionManager.java [hunk 3]
```diff
@@ -51,31 +51,33 @@
      *
      * @param server
      *            The Server your plugin is running on.
+     * @param plugin
+     *            The plugin that uses dough.
      */
     @SuppressWarnings("java:S1612")
-    public ProtectionManager(@Nonnull Server server) {
+    public ProtectionManager(@Nonnull Server server, @Nonnull Plugin plugin) {
```
Comment (WalshyDev) [inline dough-protection/src/main/java/io/github/bakedlibs/dough/protection/ProtectionManager.java:58]: Not a big fan of this being named `plugin` but can't think of a better name ccurrently

**Review by WalshyDev — APPROVED** [review]
LGTM aside one thing

**Pull request merged**